actionview/lib/action_view/lookup_context.rb

```diff
@@ -280,7 +280,15 @@ def initialize_details(target, details)
     # add :html as fallback to :js.
     def formats=(values)
       if values
+        values = values.compact
```

Comment:
Most of the quality of life improvement comes from the exception raise, so maybe we should push the check up to here as a guard clause? And then just dispense with `compact` and `uniq!` as the exception message would say: "Invalid formats: :cat, nil, :cat"

Though I don't know how likely people are to pass in either nil or non-unique formats, such that we should swallow both of them.

Reply:
`uniq!` is helpful because it will avoid duplicating formats if a user specifies ex. `[:html, "*/*"]`. 

We also do end up with `nil`s here from [_process_format](https://github.com/rails/rails/blob/master/actionview/lib/action_view/rendering.rb#L130) if fed a [Mime::NullType](https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/http/mime_type.rb#L318), but looking at that more closely, this might have changed the behaviour 🤔. I'll look into replacing the `compact` with a `dup` (I'd still like to stop mutating the argument passed in) and avoiding passing `nil` from higher up.

Reply:
Done! I've kept `uniq!`, but have replaced `compact` with `dup`.